kfs-core/src/main/java/edu/arizona/kfs/fp/document/web/struts/GeneralErrorCorrectionAction.java

```diff
@@ -377,12 +357,34 @@ public ActionForward deleteSourceLine(ActionMapping mapping, ActionForm form, Ht
     @Override
     public ActionForward deleteTargetLine(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
         GeneralErrorCorrectionForm gecForm = (GeneralErrorCorrectionForm) form;
+
+        // Get rid of GecEntryRelationship before it's removed from doc
+        GeneralErrorCorrectionDocument doc = (GeneralErrorCorrectionDocument) gecForm.getDocument();
+        deleteGecEntryRelationship(doc, doc.getTargetAccountingLine(getLineToDelete(request)));
+
+        // Now get super to do the work
         ActionForward forward = super.deleteTargetLine(mapping, form, request, response);
 
         resequenceAccountingLines(gecForm);
         return forward;
     }
 
+    private void deleteGecEntryRelationship(GeneralErrorCorrectionDocument doc, AccountingLine accountingLine) {

```

Comment:
Forgive me since I'm on a tablet and can't easily pull up all relevant code at the moment... As a general rule, I would tend to conduct the logic in place where ever the needed services/objects are already available. Said another way, if I have to pass a ton of args to the doc to make the desired changes, then I would rather treat the doc more like a bean and do the mutation in the action. If the doc does in fact have the needed services and objects already internally present, then I wholeheartedly concur encapsulation should happen there.